pennylane/transforms/decompositions/clifford_plus_t/cliffordt_transform.py

```diff
@@ -0,0 +1,272 @@
+    ]
+
+    for prod in product([pauli_terms], pauli_qubit, [pauli_terms]):
+        # hopefully op_math.prod scales better than matrix multiplication, i.e., O((2^N)^3)
+        upu = qml.pauli.pauli_sentence(qml.prod(*prod))
+        upu.simplify()
+        upu2 = upu.hamiltonian(wire_order=range(num_qubits))
+        if len(upu2.ops) == 1:
+            if not isinstance(upu2.ops[0], qml.Identity):
+                pauli_coves.append(any((qml.equal(upu2.ops[0], tm) for tm in pauli_qubit)))
+        else:
+            pauli_coves.append(False)
+
+    return all(pauli_coves)
+
+
+def _check_clifford_t(op):
+    """Checks whether the gate is in the standard Clifford+T basis"""
+    # Save time and check from the pre-computed list
+    if any(
+        (
+            isinstance(op, gate) or isinstance(getattr(op, "base", None), gate)
+            for gate in _CLIFFORD_T_GATES
+        )
+    ):
+        return True
+    return check_clifford_op(op)
+
+
+def _rot_decompose(op):
+    """Decompose a rotation operation using combination of RZ, S and Hadamard"""
+    d_ops = []
+    if isinstance(op, qml.Rot):
+        (phi, theta, omega), wires = op.data, op.wires
+        d_ops.extend(
+            [
+                qml.RZ(phi, wires),
+                qml.S(wires),
+                qml.Hadamard(wires),
+                qml.RZ(theta, wires),
+                qml.Hadamard(wires),
+                qml.adjoint(qml.S(wires)),
+                qml.RZ(omega, wires),
+            ]
+        )
+    elif isinstance(op, qml.RX):
+        (theta,), wires = op.data, op.wires
+        if theta:
+            d_ops.extend([qml.Hadamard(wires), qml.RZ(theta, wires), qml.Hadamard(wires)])
+    elif isinstance(op, qml.RY):
+        (theta,), wires = op.data, op.wires
+        if theta:
+            d_ops.extend(
+                [
+                    qml.S(wires),
+                    qml.Hadamard(wires),
+                    qml.RZ(theta, wires),
+                    qml.Hadamard(wires),
+                    qml.adjoint(qml.S(wires)),
+                ]
+            )
+    elif isinstance(op, qml.RZ):
+        (theta,), wires = op.data, op.wires
+        if theta:
+            d_ops.extend([qml.RZ(theta, wires)])
+    else:
+        d_ops.append(op)
+    return d_ops
+
+
+def _one_qubit_decompose(op):
+    """Decomposition for single qubit operations using combination of RZ and Hadamard"""
+
+    sd_ops = qml.transforms.one_qubit_decomposition(
+        qml.matrix(op), op.wires, "ZXZ", return_global_phase=True
+    )
+
+    d_ops = []
+    for sd_op in sd_ops[:-1]:
+        d_ops.extend(_rot_decompose(sd_op))
+
+    return d_ops[:-1], d_ops[-1]
+
+
+def _two_qubit_decompose(op):
+    """Decomposition for two qubit operations using combination of RZ, Hadamard, S and CNOT"""
+
+    td_ops = qml.transforms.two_qubit_decomposition(qml.matrix(op), op.wires)
+
+    d_ops = []
+    for td_op in td_ops:
+        d_ops.extend(_rot_decompose(td_op))
+
+    return d_ops
```

Comment:
I'm also not convinced on these functions, I think the standard decomposition pipeline with the right stopping condition should get us to (nearly) where we need to be, i.e., RZ, Hadamard, S, phase, and CNOT - and I'm curious to see which gates we can't do that for.

Reply:
Doing what DefaultQubit does to decompose:
```python
from pennylane.devices.qubit.preprocess import _operator_decomposition_gen as decompose
def acceptance(op):
    return op.name in {"RZ", "Hadamard", "S", "CNOT"}

op = qml.DoubleExcitation(1.1, [0, 1, 2, 3])
list(decompose(op, acceptance))
```
raises: `DeviceError: Operator RY(0.1375, wires=[1]) not supported on DefaultQubit. Must provide either a matrix or a decomposition.`

Or using what the old device would do (using `acceptance` defined above):
```pycon
>>> qs = qml.tape.QuantumScript([op])
>>> decomposed_tape = qml.tape.tape.expand_tape(qs, depth=20, stop_at=acceptance)
>>> "RY" in {o.name for o in decomposed_tape}
True
```

nothing works : )

Reply:
tl;dr the customizability of our decomposition tooling is rather immature, and I don't think we should slow down this PR because of it.

Reply:
PR #4659  is introducing `qml.devices.preprocess.decompose` as a transform, but that PR shouldn't block anything here. Also, it's for devices and not general purpose.

Reply:
> nothing works : )

We first need to add `compute_decomposition` to `RY` :+1: